internal/exec/valdiate.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/coreos/ignition/config"
+)
+
+func Validate(filename string) error {
+	b, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	_, r, err := config.Parse(b)
+	fmt.Printf("Validation report:\n%v\n", r)
+	if err != nil {
+		return err
+	}
+
+	if r.IsFatal() {
+		fmt.Println("Config had fatal errors")
```

Comment:
`fatal errors` is not the best phrasing for a user.  Maybe e.g. `Config is invalid`?